SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -0,0 +1,195 @@
+
+#include "../include/fluid/CFluidFlamelet.hpp"
+#include "../include/numerics/CLookUpTable.hpp"
+
+CFluidFlamelet::CFluidFlamelet(CConfig *config, su2double value_pressure_operating) : CFluidModel() {
+
+#ifdef HAVE_MPI
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+#endif
+
+    n_scalars = 4;
+    config->SetNScalars(n_scalars);
+
+    table_scalar_names.resize(n_scalars);
+    table_scalar_names.at(I_ENTHALPY) = "Enthalpy";
+    table_scalar_names.at(I_PROG_VAR) = "ProgVar";
+    table_scalar_names.at(I_CO)       = "Y-CO";
+    table_scalar_names.at(I_NOX)      = "Y-NOx";
+
+    config->SetScalarNames(table_scalar_names);
+
+    n_table_sources = 3;
+    config->SetNTableSources(n_table_sources);
+
+    table_source_names.resize(n_table_sources);
+    table_source_names.at(I_SRC_TOT_PROG_VAR) = "ProdRateTot-PV";
+    table_source_names.at(I_SRC_TOT_CO)       = "ProdRateTot-CO";
+    table_source_names.at(I_SRC_TOT_NOX)      = "ProdRateTot-NOx";
+
+
+    config->SetTableSourceNames(table_source_names);
+
+    look_up_table = new CLookUpTable(config->GetFileNameLUT(),table_scalar_names.at(I_PROG_VAR),table_scalar_names.at(I_ENTHALPY));
+
```

Comment:
This does not need to be instantiated with "new" unless you are planning to have polymorphic tables?
You could just have it as member variable and have a "Initialize" method rather than just the constructor.